src/config.rs

```diff
@@ -96,13 +96,16 @@ pub struct PeerConfig {
     /// If a certificate is specified here, only the specified certificate will be accepted. The system
     /// truststore will not be used.
     pub certificate: Option<String>,
+
+    pub matchkey_encryption_key: Option<String>,
```

Comment:
This won't be optional in practice, so maybe just make this a `String` directly.

Reply:
Sounds good. Although do we need to pass the matchkey public key to the peers?
I think we should also add it to ServerConfig. Will do that

Reply:
The other peers won't need this, but it probably makes sense to reuse the same configuration (or at least a subset of it) for client configuration.

Reply:
Yes, I hope ServerConfig is the right place to put it.

Reply:
I think `ServerConfig` indeed needs both the private key and the public key (in some formats, the private key is stored with a copy of the public key, but if we're just using raw bytes, that won't be the case).

I also think that the public key should go in `NetworkConfig` (`network.toml`). The idea is that `network.toml` gets distributed to clients of the helper network. `test_mpc` already takes an argument `--network network.toml`, and the event generation stuff that @akoshelev is adding to it will soon (possibly immediately, if configuration/use of the report key is mandatory) need to support encrypting the reports.

Reply:
Within NetworkConfig we have a PeerConfig. Should we add a ClientConfig which will store the public key needed for encryption? There might be other information which would be needed to communicated to clients which can go there. @andyleiserson

Reply:
I think the public key could go directly in PeerConfig. The only significance that I had in mind for `PeerConfig` was that it was a container for the configuration that needed to be replicated 3x for the helpers.

Another possibility is to represent the key information in the OHTTP key configuration format mentioned in https://github.com/patcg-individual-drafts/ipa/blob/main/details/encryption.md#helper-key-configuration, but that doesn't seem important to do right now, and even using that format I think we could nest it within PeerConfig.

Reply:
There are a few things in PeerConfig (e.g. port) which advertisers/publishers don't care about. I will move public_key inside PeerConfig and then we can see how it evolves